AddOns/BasicFinancials/app/src/codeunit/BasicFinancialsMgmt.Codeunit.al

```diff
@@ -0,0 +1,121 @@
+        BFSKUIdTxt: Label '{66CAD104-64F9-476E-9682-3C3518B9B6ED}', Locked = true, Comment = 'Dynamics 365 Business Central Basic Financials';
+        BFC5SPLASKUIdTxt: Label '{4dCE07FD-7B07-4FB5-8FB7-D49653F7BF30}', Locked = true, Comment = 'Dynamics 365 Business Central Basic Financials from C5 SPLA (Qualified Offer)';
+
+    internal procedure IsSupportedLicense(): Boolean // A microsoft requirements: The Basic Financials Assisted Setup checks for the Basic Financials license on the AAD tenant, at least one user has been assigned to this license.
+    var
+    begin
+        AzureADLicensing.ResetSubscribedSKU();
+        while AzureADLicensing.NextSubscribedSKU() do
+            case UpperCase(AzureADLicensing.SubscribedSKUId()) of
+                BFSKUIdTxt:
+                    exit(true);
+                BFC5SPLASKUIdTxt:
+                    exit(true);
+            end;
+        exit(false);
+    end;
+
+    internal procedure TestSupportedLicenses() // A microsoft requirements: The extension checks for the Basic Financials license on the AAD tenant, at least one user has been assigned to this license. 
+    var
+    begin
+        AzureADLicensing.ResetSubscribedSKU();
+        while AzureADLicensing.NextSubscribedSKU() do
+            case UpperCase(AzureADLicensing.SubscribedSKUId()) of
+                BFSKUIdTxt:
+                    exit;
+                BFC5SPLASKUIdTxt:
+                    exit;
+            end;
+        Error(NotSupportedLicenseErr);
+    end;
+
+    internal procedure TestSupportedLocales() // A microsoft requirements: The extension checks for the country availability, the extension is only available to the  Countries: Denmark, Iceland';
+    var
+        TempApplicationAreaSetup: Record "Application Area Setup";
+        AppAreaMgmt: Codeunit "App Area Mgmt BF";
+
+    begin
+        Clear(TempApplicationAreaSetup);
+        AppAreaMgmt.GetEssentialExperienceAppAreas(TempApplicationAreaSetup);
+        if TempApplicationAreaSetup."Basic DK" or TempApplicationAreaSetup."Basic IS" then
+            exit;
+
+        Error(NotSupportedLocalesErr);
+    end;
+
+    internal procedure TestSupportedUser() // A microsoft requirements: The extension checks for User Permissions, the user which install the extension has been assigned to Super. 
+    var
+        User: Record User;
+        UserPermissions: Codeunit "User Permissions";
+    begin
+        User.Get(UserSecurityId()); // Something unexpected happened while trying to get the current user, it will raise an exception (error message)
+
+        if User.IsEmpty() then
+            exit;
+
+        if UserPermissions.IsSuper(UserSecurityId()) then
+            exit;
+
+        Error(NotSupportedUserErr, User."User Name");
+    end;
+
+    internal procedure TestSupportedCompanies() // A microsoft requirements: The extension checks for only 1 company is installed on the tenant, additional companies has to be added afterward. 
```

Comment:
Microsoft requirements: The extension checks whether the tenant contains more than one company during installation, but allows additional companies to be added afterward.